app/code/Magento/NewsletterGraphQl/Model/Resolver/SubscribeEmailToNewsletter.php

```diff
@@ -0,0 +1,237 @@
+
+    /**
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfig;
+
+    /**
+     * @var SubscriberResourceModel
+     */
+    private $subscriberResource;
+
+    /**
+     * @var SubscriptionManagerInterface
+     */
+    private $subscriptionManager;
+
+    /**
+     * SubscribeEmailToNewsletter constructor.
+     *
+     * @param CustomerAccountManagement $customerAccountManagement
+     * @param CustomerRepositoryInterface $customerRepository
+     * @param EmailValidator $emailValidator
+     * @param EnumLookup $enumLookup
+     * @param ScopeConfigInterface $scopeConfig
+     * @param SubscriberResourceModel $subscriberResource
+     * @param SubscriptionManagerInterface $subscriptionManager
+     */
+    public function __construct(
+        CustomerAccountManagement $customerAccountManagement,
+        CustomerRepositoryInterface $customerRepository,
+        EmailValidator $emailValidator,
+        EnumLookup $enumLookup,
+        ScopeConfigInterface $scopeConfig,
+        SubscriberResourceModel $subscriberResource,
+        SubscriptionManagerInterface $subscriptionManager
+    ) {
+        $this->customerAccountManagement = $customerAccountManagement;
+        $this->customerRepository = $customerRepository;
+        $this->emailValidator = $emailValidator;
+        $this->enumLookup = $enumLookup;
+        $this->scopeConfig = $scopeConfig;
+        $this->subscriberResource = $subscriberResource;
+        $this->subscriptionManager = $subscriptionManager;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $email = trim($args['email']);
+
+        if (empty($email)) {
+            throw new GraphQlInputException(__('You must specify an email address to subscribe to a newsletter.'));
+        }
+
+        try {
+            $currentUserId = (int)$context->getUserId();
+            $storeId = (int)$context->getExtensionAttributes()->getStore()->getId();
+            $websiteId = (int)$context->getExtensionAttributes()->getStore()->getWebsiteId();
+
+            $this->validateEmailFormat($email);
+            $this->validateGuestSubscription($context);
+            $this->validateEmailAvailable($email, $currentUserId, $websiteId);
+            $this->validateAlreadySubscribed($email, $websiteId);
+
+            $subscriber = $this->isCustomerSubscription($email, $currentUserId)
+                ? $this->subscriptionManager->subscribeCustomer($currentUserId, $storeId)
+                : $this->subscriptionManager->subscribe($email, $storeId);
+
+            $status = $this->enumLookup->getEnumValueFromField(
+                'SubscriptionStatusesEnum',
+                (string)$subscriber->getSubscriberStatus()
+            );
+        } catch (LocalizedException $e) {
+            throw new GraphQlInputException(__($e->getMessage()));
```

Comment:
Please do not show `$e->getMessage()`. It may cause security issue.
Message should be `Cannot create newsletter subscription`. Exception message should be logged.

Reply:
Also I would avoid creating private methods and the re-catch exceptions.

Reply:
**done**